core/src/main/resources/hudson/model/Job/configure.jelly

```diff
@@ -56,6 +56,7 @@ THE SOFTWARE.
           <f:entry title="${%Description}" help="${app.markupFormatter.helpUrl}">
             <f:textarea name="description" value="${it.description}" codemirror-mode="${app.markupFormatter.codeMirrorMode}" codemirror-config="${app.markupFormatter.codeMirrorConfig}" previewEndpoint="/markupFormatter/previewDescription"/>
           </f:entry>
+          <f:dropdownDescriptorSelector title="${%Icon}" field="icon" />
```

Comment:
this should probably only be shown if there's more than one implementation available, there should be examples of that around for the console url provider

Reply:
Done in https://github.com/jenkinsci/jenkins/pull/9723/commits/6b519f4b4021d456e2a0391a7f501e22f8c8fd7d